ta4j-core/src/main/java/org/ta4j/core/indicators/averages/WildersMAIndicator.java

```diff
@@ -0,0 +1,111 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.averages;
+
+import org.ta4j.core.Indicator;
+import org.ta4j.core.indicators.CachedIndicator;
+import org.ta4j.core.num.Num;
+import org.ta4j.core.num.NumFactory;
+
+/**
+ * Wilder's moving average indicator.
+ *
+ * The Wilder’s Moving Average indicator (Wilder’s Smoothed Moving Average ) was
+ * developed by Welles Wilder and introduced in his 1978 book, “New Concepts in
+ * Technical Trading Systems.”
+ *
+ * The WMA inherently "remembers" past values due to its recursive formula, but
+ * the initialization WMA can significantly bias early values. This bias
+ * diminishes over time as more data contributes to the calculation. See the
+ * test cases for this indicator as it skips the initial data until the
+ * calculation stabilizes.
+ *
+ * @see <a href=
+ *      "https://www.tradingview.com/script/wXtQeoOg/#:~:text=Wilder%20did%20not%20use%20the,where%20K%20%3D1%2FN.">
+ *      https://www.tradingview.com/script/wXtQeoOg/#:~:text=Wilder%20did%20not%20use%20the,where%20K%20%3D1%2FN.</a>
+ */
+public class WildersMAIndicator extends CachedIndicator<Num> {
+
+    private final Indicator<Num> indicator;
+    private final int barCount;
+
+    /**
+     * Constructor.
+     *
+     * @param indicator an indicator
+     * @param barCount  the Wilder Moving Average time frame
+     */
+    public WildersMAIndicator(Indicator<Num> indicator, int barCount) {
+        super(indicator);
+        this.indicator = indicator;
+        this.barCount = barCount;
+    }
+
+    @Override
+    protected Num calculate(int index) {
+        NumFactory numFactory = getBarSeries().numFactory();
+        Num one = numFactory.numOf(1);
+
+        Num k = one.dividedBy(numFactory.numOf(barCount));
+        Num prevWMA = indicator.getValue(0);
+
+        // Simulate extended historical data for initialization
+        if (index < barCount) {
+            // Pretend there are extra `barCount` points before the first real point
+            // Use the first actual value as a proxy for the "missing" earlier data
+            Num simulatedValue = indicator.getValue(0);
+
+            // Initialize with a simulated Simple Moving Average (SMA)
+            Num sum = numFactory.zero();
+            for (int i = 0; i <= index; i++) {
```

Comment:
Is it possible to reuse RunningTotalIndicator?

Reply:
Changed to RunningTotalIndicator